File: zapit-backend/src/main/java/com/shaunwah/zapitbackend/model/Transaction.java
package com.shaunwah.zapitbackend.model;

import lombok.*;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    private Long id;
    private Card card;
    private Double amount;
    private Boolean status;
    private LocationData location;
    private Timestamp createdOn;
    private Timestamp updatedOn;
}

File: zapit-frontend/src/app/transaction/transaction.ts
import { Card } from '../card/card';
import { LocationData } from '../shared/interfaces/location-data';

export interface Transaction {
  id?: number;
  card: Card;
  amount: number;
  status: boolean;
  location?: LocationData;
  createdOn?: number;
  updatedOn?: number;
}
